maidr/custom_renderers/pyshiny_renderer.py

```diff
@@ -0,0 +1,48 @@
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+import maidr
+from shiny import ui
+from typing import Any
+
+
+def render_maidr(func):
+    def wrapper(*args, **kwargs):
+        try:
+            plot_obj = func(*args, **kwargs)
+            rendered_dict = maidr.render(plot_obj)
+
+            html_content = rendered_dict.get("html", "")
+            dependencies = rendered_dict.get("dependancies", [])
+
+            # Create a string to hold all dependency tags
+            dependency_tags = ""
+            for dep in dependencies:
+                if dep.get("type") == "script":
+                    dependency_tags += f'<script src="{dep["src"]}"></script>'
+                elif dep.get("type") == "stylesheet":
+                    dependency_tags += f'<link rel="stylesheet" href="{dep["href"]}">'
+
+            # Combine dependencies and plot HTML
+            full_html_content = f"""
+            <!DOCTYPE html>
+            <html>
+            <head>
+                {dependency_tags}
+            </head>
+            <body>
+                {html_content}
+            </body>
+            </html>
+            """
+
```

Comment:
any reasons why you don't use htmltools package and its tag methods instead of the direct string literals for html tags?

Reply:
> During the initial development of the feature, I was able to load the entire maidr object on to the shiny dashboard component but was unable to make use of maidr's interactiveness. Upon analyzing the scripts, I realized that maidr's javascript objects were being loaded on to the webpage but shiny was preventing me from using it.
> 
> In order to have control over the HTML component I am feeding into iframe, I decided to use generic string structures for the HTML tags instead of html tools.
> 
> With the current implementation as a viable fallback, I will begin the migration to htmltools for this section.

Have you taken a look at this method by chance?: https://shiny.posit.co/py/api/core/Htmltools.html